### PR TITLE
MAGECLOUD-2870: Static assets compression shouldn't break a deploy

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description
Added possibility for configuration timeout time for compression of static files. Previously it was hardcoded as 600s and it wasn't enough in some cases.
Removed from constructors creation of connection data objects via `ConnectionFactory` and moved to methods where it required.

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-2870

### Manual testing scenarios

https://jira.corp.magento.com/browse/MAGECLOUD-28
### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Config/Database/MergedConfig.php [hunk 1]
```diff
@@ -96,7 +103,7 @@ public function get(): array
             return $this->mergedConfig = $this->configMerger->clear($envDbConfig);
         }
 
-        if (!empty($this->connectionData->getHost())) {
+        if (!empty($this->getConnectionData()->getHost())) {
```
Comment (shiftedreality) [inline src/Config/Database/MergedConfig.php:106]: `if ($value)` should be enough, `!empty()` does not bring additional value

File: src/DB/Connection.php [hunk 2]
```diff
@@ -184,14 +183,15 @@ private function connect()
             return;
         }
 
+        $connectionData = $this->connectionFactory->create(ConnectionFactory::CONNECTION_MAIN);
```
Comment (shiftedreality) [inline src/DB/Connection.php:186]: Nice catch

File: src/DB/Dump.php [hunk 3]
```diff
@@ -36,19 +35,20 @@ public function __construct(
      */
     public function getCommand(): string
     {
-        $command = 'mysqldump -h ' . escapeshellarg($this->connectionData->getHost())
-            . ' -u ' . escapeshellarg($this->connectionData->getUser());
+        $connectionData = $this->connectionFactory->create(ConnectionFactory::CONNECTION_SLAVE);
```
Comment (shiftedreality) [inline src/DB/Dump.php:38]: Don't we want to share a connection with `get()`?

**Review by shiftedreality — APPROVED** [review]

File: src/Config/Database/MergedConfig.php (hunk 1 above)
Comment (oshmyheliuk) [reply src/Config/Database/MergedConfig.php:106]: Yeah, you right, but I didn't want to change existence logic

File: src/DB/Dump.php (hunk 3 above)
Comment (oshmyheliuk) [reply src/DB/Dump.php:38]: Sorry, don't udnerstand the question. Can you clarify please?

Comment (shiftedreality) [reply src/DB/Dump.php:38]: Like `make` always create a new instance of connection i.e. opens a new connection. `get` would share this as singleton without creating a new one if it's already exists

Comment (oshmyheliuk) [reply src/DB/Dump.php:38]: The configuration actually cached in MergedConfig. So there no additional opertion for creating new instance of Connection.

File: src/Config/Schema.php [hunk 4]
```diff
@@ -96,6 +96,18 @@ public function getSchema()
...
             ],
+            StageConfigInterface::VAR_SCD_COMPRESSION_TIMEOUT => [
+                self::SCHEMA_TYPE => ['integer'],
+                self::SCHEMA_STAGE => [
+                    StageConfigInterface::STAGE_GLOBAL,
+                    StageConfigInterface::STAGE_BUILD,
+                    StageConfigInterface::STAGE_DEPLOY
+                ],
+                self::SCHEMA_DEFAULT_VALUE => [
+                    StageConfigInterface::STAGE_BUILD => 600,
```
Comment (NadiyaS) [inline src/Config/Schema.php:107]: Why don't we use constant with default values?

**Review by NadiyaS — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA passed

**Pull request merged**